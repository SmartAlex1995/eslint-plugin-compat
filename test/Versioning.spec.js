import DetermineTargetsFromConfig, { Versioning } from '../src/Versioning';
import multiEnvPackageJSON from './multi-config.package.json';
import singleArrayEnvPackageJSON from './single-array-config.package.json';
import singleVersionEnvPackageJSON from './single-version-config.package.json';


describe('Versioning', () => {
  it('should support multi env config in browserslist package.json', () => {
    const config = DetermineTargetsFromConfig(multiEnvPackageJSON.browsers);
    const result = Versioning(config);
    expect(result).toMatchSnapshot();
  });

  it('should support single array config in browserslist package.json', () => {
    const config = DetermineTargetsFromConfig(singleArrayEnvPackageJSON.browsers);
    const result = Versioning(config);
    expect(result).toMatchSnapshot();
  });

  it('should support single version config in browserslist package.json', () => {
    const config = DetermineTargetsFromConfig(singleVersionEnvPackageJSON.browsers);
    const result = Versioning(config);
    expect(result).toMatchSnapshot([
      { target: 'safari', version: '8', parsedVersion: 8 },
      { target: 'ie', version: '9', parsedVersion: 9 },
      { target: 'firefox', version: '20', parsedVersion: 20 },
      { target: 'chrome', version: '32', parsedVersion: 32 }
    ]);
  });

  it('should get lowest target versions', () => {
    const versions = [
      'chrome 20',
      'chrome 30',
      'node 7',
      'chrome 30.5',
      'firefox 50.5'
    ];
    expect(Versioning(versions)).toMatchSnapshot();
  });
});
