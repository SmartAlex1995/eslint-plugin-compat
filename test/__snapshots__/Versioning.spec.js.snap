// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Versioning should get lowest target versions 1`] = `
Array [
  Object {
    "parsedVersion": 7,
    "target": "node",
    "version": "7",
  },
  Object {
    "parsedVersion": 50.5,
    "target": "firefox",
    "version": "50.5",
  },
  Object {
    "parsedVersion": 20,
    "target": "chrome",
    "version": "20",
  },
]
`;

exports[`Versioning should support multi env config in browserslist package.json 1`] = `
Array [
  Object {
    "parsedVersion": 4,
    "target": "samsung",
    "version": "4",
  },
  Object {
    "parsedVersion": 10,
    "target": "safari",
    "version": "10",
  },
  Object {
    "parsedVersion": 50,
    "target": "opera",
    "version": "50",
  },
  Object {
    "parsedVersion": 11.5,
    "target": "op_mob",
    "version": "11.5",
  },
  Object {
    "parsedVersion": 0,
    "target": "op_mini",
    "version": "all",
  },
  Object {
    "parsedVersion": 10,
    "target": "ios_saf",
    "version": "10.0-10.2",
  },
  Object {
    "parsedVersion": 10,
    "target": "ie_mob",
    "version": "10",
  },
  Object {
    "parsedVersion": 9,
    "target": "ie",
    "version": "9",
  },
  Object {
    "parsedVersion": 52,
    "target": "firefox",
    "version": "52",
  },
  Object {
    "parsedVersion": 14,
    "target": "edge",
    "version": "14",
  },
  Object {
    "parsedVersion": 63,
    "target": "chrome",
    "version": "63",
  },
  Object {
    "parsedVersion": 7,
    "target": "bb",
    "version": "7",
  },
  Object {
    "parsedVersion": 7.12,
    "target": "baidu",
    "version": "7.12",
  },
  Object {
    "parsedVersion": 4.2,
    "target": "android",
    "version": "4.2-4.3",
  },
  Object {
    "parsedVersion": 11.8,
    "target": "and_uc",
    "version": "11.8",
  },
  Object {
    "parsedVersion": 1.2,
    "target": "and_qq",
    "version": "1.2",
  },
  Object {
    "parsedVersion": 60,
    "target": "and_ff",
    "version": "60",
  },
  Object {
    "parsedVersion": 66,
    "target": "and_chr",
    "version": "66",
  },
]
`;

exports[`Versioning should support single array config in browserslist package.json 1`] = `
Array [
  Object {
    "parsedVersion": 4,
    "target": "samsung",
    "version": "4",
  },
  Object {
    "parsedVersion": 10,
    "target": "safari",
    "version": "10",
  },
  Object {
    "parsedVersion": 50,
    "target": "opera",
    "version": "50",
  },
  Object {
    "parsedVersion": 11.5,
    "target": "op_mob",
    "version": "11.5",
  },
  Object {
    "parsedVersion": 0,
    "target": "op_mini",
    "version": "all",
  },
  Object {
    "parsedVersion": 10,
    "target": "ios_saf",
    "version": "10.0-10.2",
  },
  Object {
    "parsedVersion": 10,
    "target": "ie_mob",
    "version": "10",
  },
  Object {
    "parsedVersion": 9,
    "target": "ie",
    "version": "9",
  },
  Object {
    "parsedVersion": 52,
    "target": "firefox",
    "version": "52",
  },
  Object {
    "parsedVersion": 14,
    "target": "edge",
    "version": "14",
  },
  Object {
    "parsedVersion": 57,
    "target": "chrome",
    "version": "57",
  },
  Object {
    "parsedVersion": 7,
    "target": "bb",
    "version": "7",
  },
  Object {
    "parsedVersion": 7.12,
    "target": "baidu",
    "version": "7.12",
  },
  Object {
    "parsedVersion": 4.2,
    "target": "android",
    "version": "4.2-4.3",
  },
  Object {
    "parsedVersion": 11.8,
    "target": "and_uc",
    "version": "11.8",
  },
  Object {
    "parsedVersion": 1.2,
    "target": "and_qq",
    "version": "1.2",
  },
  Object {
    "parsedVersion": 60,
    "target": "and_ff",
    "version": "60",
  },
  Object {
    "parsedVersion": 66,
    "target": "and_chr",
    "version": "66",
  },
]
`;

exports[`Versioning should support single version config in browserslist package.json 1`] = `
Array [
  Object {
    "parsedVersion": 8,
    "target": "safari",
    "version": "8",
  },
  Object {
    "parsedVersion": 9,
    "target": "ie",
    "version": "9",
  },
  Object {
    "parsedVersion": 20,
    "target": "firefox",
    "version": "20",
  },
  Object {
    "parsedVersion": 32,
    "target": "chrome",
    "version": "32",
  },
]
`;
